### DISCUSSION TITLE
afterFind Hook Update Problem

### DISCUSSION THREAD

**Discussion by atakansn** [adonisjs/core#4727] [Help]
```
@afterFind() 
public static async serveProfilePhotoHook(user: User) { 
  if(!user.profilePhotoPath) { 
   const name = user.fullName.trim().split(' ').map((name: string) => {
     return name.charAt(0) 
   }).join(' ') 

   user.profilePhotoPath = https://ui-avatars.com/api/?name=${encodeURIComponent(name)}&color=7F9CF5&background=EBF4FF

   return 
  } 

  user.profilePhotoPath = await drive.use().getUrl(user.profilePhotoPath) 
}
```

my problem is that when the user updates any data, user.profilePhotoPath also updates, whereas I just want to present the image to the user asyncly

**Comment by shazzsamed** [comment]
This hook is invoked every time you use the find method. So if you are updating after using the find method it might update profile photo every time.
Correct me if i am wrong

**Comment by RomainLanz** [comment]
Hey @atakansn! 👋🏻 

Could you explain exactly your needs? It seems to be an [XY Problem](https://xyproblem.info/) to me.
